memoization.js

```diff
@@ -1,38 +1,26 @@
 'use strict';
 
-const smartSelector = (() => {
-  const getElementById = ((id) => {
-    let cache = {};
+const smartSelect = (() => {
+  const cache = {};
 
-    return (id) => {
-      if (id in cache) {
-        return cache[id];
-      } else {
-        let select = document.getElementById(id);
+  const querySelector = (query) => {
+    const format = query.charAt(0) === '#' ? query : `#${query}`;
+    const selection = document.querySelector(format);
 
-        cache[id] = select;
-        return select;
-      }
+    if (format in cache) {
+      return cache[format];
     }
-  })();
 
-  const querySelector = ((query) => {
-    let cache = {};
-
-    return (query) => {
-      if (query in cache) {
-        return cache[query];
-      } else {
-        let select = document.querySelector(query);
+    cache[format] = selection;
+    return selection;
+  }
 
-        cache[query] = select;
-        return select;
-      }
-    }
-  })();
-  
   return {
-    superGetId : getElementById,
     superQuery : querySelector
```

Comment:
I would say that you should probably use a name closer to what the original methods you're trying to memoize are doing. Like `superQuerySelectorAll`at least.